bin/dasht-docsets-extract

```diff
@@ -50,32 +60,50 @@
 # Written in 2016 by Suraj N. Kurapati <https://github.com/sunaku/dasht>
 # Distributed under the terms of the ISC license (refer to README file).
 
-: ${DASHT_DOCSETS_DIR:=${XDG_DATA_HOME:-$HOME/.local/share}/dasht/docsets}
+: ${DASHT_USE_DASH_APP:=false}
+if $DASHT_USE_DASH_APP; then
+  : ${DASHT_DOCSETS_DIR:="/Users/smackesey/Library/Application Support/Dash"}
+else
+  : ${DASHT_DOCSETS_DIR:=${XDG_DATA_HOME:-$HOME/.local/share}/dasht/docsets}
+fi
 
 test $# -gt 0
 for tgz; do
-  want="$DASHT_DOCSETS_DIR/$(basename "$tgz" .tgz).docset"
-  have="$DASHT_DOCSETS_DIR/$(tar -t -f "$tgz" -z | head -1 | sed 's|/.*||')"
 
-  # for dasht-docsets-update(1), the extracted directory will already exist
-  # with the correct name, but we need to temporarily rename it back to the
-  # wrong name so that the archive extraction below can update its contents
-  # in place, rather than extracting out to a new (and different) directory
-  test "$have" = "$want" -o ! -e "$want" || mv "$want" "$have"
+  # Extract Documents dir inside existing Dash.app docset directory. Dash.app
+  # leaves the Documents inside the tarball tarix.tgz by default.
+  if $DASHT_USE_DASH_APP; then
+    pushd "$(dirname "$tgz")" >/dev/null
+    docset_name="$(basename "$(dirname "$(dirname "$(dirname "$tgz")")")")"
```

Comment:
Look at [POSIX `sh` Parameter Expansion](https://pubs.opengroup.org/onlinepubs/9699919799/utilities/V3_chap02.html#tag_18_06_02) and check if the `${var%%...}` operator is applicable.